packages/component-base/test/focus-utils.test.js

```diff
@@ -0,0 +1,154 @@
+      const element = document.createElement('iframe');
+      expect(isElementFocusable(element)).to.be.true;
+    });
+
+    it('should return false for <iframe> with [tabindex] = -1', () => {
+      const element = document.createElement('iframe');
+      element.setAttribute('tabindex', '-1');
+      expect(isElementFocusable(element)).to.be.false;
+    });
+
+    it('should return false for <div> by default', () => {
+      const element = document.createElement('div');
+      expect(isElementFocusable(element)).to.be.false;
+    });
+
+    it('should return true for <div> with [tabindex] >= 0', () => {
+      const element = document.createElement('div');
+      element.setAttribute('tabindex', '0');
+      expect(isElementFocusable(element)).to.be.true;
+    });
+
+    it('should return true for <div> with [contenteditable]', () => {
+      const element = document.createElement('div');
+      element.setAttribute('contenteditable', '');
+      expect(isElementFocusable(element)).to.be.true;
+    });
+
+    it('should return false for <div> with [contenteditable] and [tabindex] = -1', () => {
+      const element = document.createElement('div');
+      element.setAttribute('contenteditable', '');
+      element.setAttribute('tabindex', '-1');
+      expect(isElementFocusable(element)).to.be.false;
+    });
+  });
+
+  describe('getFocusableElements', () => {
+    let root;
+
+    customElements.define(
+      'custom-element',
+      class extends PolymerElement {
+        static get template() {
+          return html`
+            <div id="custom-element-1" tabindex="0"></div>
+            <div id="custom-element-2" tabindex="-1"></div>
+            <slot>
+              <div id="custom-element-3" tabindex="3"></div>
+            </slot>
+          `;
+        }
+      }
+    );
+
+    beforeEach(() => {
+      root = fixtureSync(`
+        <div id="root" tabindex="0">
+          content
+          <button id="element-1"></button>
+          <button id="element-2" tabindex="-1"></button>
+          <select id="element-3" tabindex="2">
+            <option>tabindex 2</option>
+          </select>
+          <textarea id="element-4" tabindex="1"></textarea>
+          <input type="text" id="element-5" />
+          <input type="text" id="element-6" style="display: none;" />
+          <input type="text" id="element-7" style="visibility: hidden;" />
+          <input type="radio" id="element-8" />
+          <input type="checkbox" id="element-9" />
+          <custom-element></custom-element>
+        </div>
+      `);
+    });
+
+    it('should return focusable elements in the tab order', () => {
+      const focusableElements = getFocusableElements(root);
+      expect(focusableElements.map((element) => element.id)).to.deep.equal([
+        'element-4',
+        'element-3',
+        'custom-element-3',
+        'root',
+        'element-1',
+        'element-5',
+        'element-8',
+        'element-9',
+        'custom-element-1'
```

Comment:
Probably not a regression from this PR, (but rather a flaw in the original implementation,) but for me the natural tab order with this setup was:
```
element-4
element-3
root
element-1
element-5
element-8
element-9
custom-element-1
custom-element-3
```

Reply:
I double-checked it and yeah, it used to work the same in the original helper so that it is not a regression.

Would you like to create an issue about this? Although, I'm not sure if it is a serious problem as long as we use this order for custom Tab navigation. Well, the order will be slightly different but that's all.

Reply:
I can create a separate issue once this is merged